Override system libsecp256k1 for coincurve

fivepiece opened the pull request:
This is for testing a possible fix for #234

undeath commented:
doesn't work for me

    >>> import jmbitcoin
    Traceback (most recent call last):
      File "<console>", line 1, in <module>
      File "/home/test/joinmarket-clientserver/jmvenv/lib/python2.7/site-packages/jmbitcoin/__init__.py", line 1, in <module>
        import coincurve as secp256k1
      File "/home/test/joinmarket-clientserver/jmvenv/lib/python2.7/site-packages/coincurve-9.0.0-py2.7-linux-x86_64.egg/coincurve/__init__.py", line 1, in <module>
        from coincurve.context import GLOBAL_CONTEXT, Context
      File "/home/test/joinmarket-clientserver/jmvenv/lib/python2.7/site-packages/coincurve-9.0.0-py2.7-linux-x86_64.egg/coincurve/context.py", line 4, in <module>
        from coincurve.flags import CONTEXT_ALL, CONTEXT_FLAGS
      File "/home/test/joinmarket-clientserver/jmvenv/lib/python2.7/site-packages/coincurve-9.0.0-py2.7-linux-x86_64.egg/coincurve/flags.py", line 1, in <module>
        from ._libsecp256k1 import lib
    ImportError: /home/test/joinmarket-clientserver/jmvenv/lib/python2.7/site-packages/coincurve-9.0.0-py2.7-linux-x86_64.egg/coincurve/_libsecp256k1.so: undefined symbol: secp256k1_ecdh

I guess coincurve is still trying to use the system lib after compiling against the local one.

AdamISZ commented:
tACK from me (I did: wipe any jmvenv/ and deps/ ; run install.sh --develop; source jmvenv/bin/activate; run test suite).

fivepiece commented:
@undeath can you paste the output from
```
strings jmvenv/lib/python2.7/site-packages/coincurve-9.0.0-py2.7-linux-x86_64.egg/coincurve/_libsecp256k1.so | grep "\.h$" | sort -u
```

fivepiece commented:
Sorry all, I had to force push.  Accidentally rebased an incorrect commit.
@undeath I managed to recreate the same error you're getting locally on a gentoo docker and I think it should be fixed now.

AdamISZ commented:
pinging @AlexCato , does this fix your installation?

undeath commented:
I wonder if this problem I encountered is a problem with coincurve's build system. It looks like they try to add it to the linker options but for some reason it doesn't work.

https://github.com/ofek/coincurve/blob/972912506f6ffb293a1d76311dbb162cfb50a909/setup.py#L200

fivepiece commented:
I changed `install.sh` to be more inline with ofek/coincurve#36 and removed comments that were referencing a previous issue.  Ready to squash when needed.

undeath commented:
ACK
